cms/djangoapps/contentstore/views/import_export.py

```diff
@@ -370,7 +373,10 @@ def export_status_handler(request, course_key_string):
     elif task_status.state == UserTaskStatus.SUCCEEDED:
         status = 3
         artifact = UserTaskArtifact.objects.get(status=task_status, name='Output')
-        if hasattr(artifact.file.storage, 'bucket'):
+        if isinstance(artifact.file.storage, FileSystemStorage):
+            # local file, serve from the authorization wrapper view
+            output_url = reverse_course_url('export_output_handler', course_key)
+        elif isinstance(artifact.file.storage, S3BotoStorage):
```

Comment:
I am somewhat reading between the lines here, but I assume that checking for the bucket attribute is meant to be a way to check for S3 storage without explicitly importing the S3BotoStorage class.  

Checking for the ``bucket`` attribute will match any storage that has the bucket attribute, which includes [S3BotoStorage](https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto.py#L270). 

Another hint is that if the bucket attribute is found, it calls the ``url`` method of the storage with the extra ``response_headers`` parameter, which is not part of the standard Django storage contract but is supported by [S3BotoStorage](https://github.com/jschneier/django-storages/blob/9f07eab9c2f86c3d911ec9f6b0f45dca36626bd1/storages/backends/s3boto.py#L470). 

I have not tested this with S3BotoStorage, but I have set up sandboxes to test with FileSystemStorage and Swift storage (for the [upstream PR](/edx/edx-platform#17900) ).

I did not check for the S3BotoStorage case since that seemed the most straightforward. The only thing this could potentially affect is if someone uses a different storage solution that has the bucket parameter, but isn't ``S3BotoStorage``, and doesn't subclass from it.

Reply:
Note: I can revert to the ``hasattr`` approach for S3, but I just changed it because this seemed like the cleaner approach to me, and ``S3BotoStorage`` is available anyway, even if you aren't hosting on AWS.

Reply:
It would need to have ``bucket`` and also support ``response_headers`` in ``url``, otherwise it would throw an exception, so that narrows things even more!